src/core/hooks/useIsAdmin.ts

```diff
@@ -0,0 +1,7 @@
+import { UserDTO } from '../models/dto/auth.dto';
+
+export const useIsAdmin = (user: UserDTO) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const isAdmin = (user as any)?.roles[0].name === 'SuperAdmin';
```

Comment:
according to this, the user can be empty or it is possible that it doesn't have the `roles` property, so you should make sure that the `roles` property is actually an array, has at least 1 element and that element has the `name` property

Could we have an enum with the possible roles?

Reply:
Yes we can move the role to Enum, I will fix the useIsAdmin, because the way to determine that is through the permissions, I will make the change